examples/stocks/lib/stock_home.dart

```diff
@@ -24,12 +24,15 @@ class StockHomeState extends State<StockHome> {
   String _searchQuery;
 
   void _handleSearchBegin() {
-    Navigator.of(context).pushState(this, (_) {
-      setState(() {
-        _isSearching = false;
-        _searchQuery = null;
-      });
-    });
+    Navigator.of(context).push(new StateRoute(
+      owner: this,
```

Comment:
we only use the owner in a single assert. Let's drop it.

That leaves this as being just a single callback. That makes sense. It's basically saying, "we're doing something, here's what to do when they hit 'back'".